cdm/core/src/main/java/ucar/nc2/internal/iosp/netcdf3/N3iospNew.java

```diff
@@ -406,6 +422,26 @@ public Object sendIospMessage(Object message) {
     return super.sendIospMessage(message);
   }
 
+  /**
+   * Return {@link Charset value charset} if it was defined. Definition of charset
+   * occurs by sending a charset as a message using the {@link #sendIospMessage}
+   * method.
+   * 
+   * @return {@link Charset value charset} if it was defined.
+   */
+  protected Optional<Charset> getValueCharset() {
+    return Optional.ofNullable(valueCharset);
+  }
+
+  /**
+   * Define {@link Charset value charset}.
+   * 
+   * @param charset may be null.
+   */
+  protected void setValueCharset(Charset charset) {
```

Comment:
Would you mind annotating this method's signature with `@Nullable` from `javax.annotation`, so something like `setValueCharset(@Nullable Charset charset)`?